mesecons/spec/service_spec.lua

```diff
@@ -0,0 +1,186 @@
+require("mineunit")
+
+fixture("mesecons")
+fixture("screwdriver")
+
+describe("placement/digging service", function()
+	local layout = {
+		{{x =  1, y = 0, z = 0}, "mesecons:test_receptor_on"},
+		{{x =  0, y = 0, z = 0}, "mesecons:test_conductor_on"},
+		{{x = -1, y = 0, z = 0}, "mesecons:test_conductor_on"},
+		{{x =  0, y = 1, z = 0}, "mesecons:test_effector"},
+		{{x = -2, y = 0, z = 0}, "mesecons:test_effector"},
+		{{x =  2, y = 0, z = 0}, "mesecons:test_effector"},
+	}
+
+	before_each(function()
+		world.layout(layout)
+	end)
+
+	after_each(function()
+		mesecon._test_reset()
+		world.clear()
+	end)
+
+	it("updates components when a receptor changes", function()
+		mesecon._test_dig(layout[1][1])
```

Comment:
```suggestion
		-- Remove the source (receptor) and check whether the adjacent conductor(s) turn off
		mesecon._test_dig(layout[1][1])
```

It would be good to have a brief comment for some of the more complicated setups. Otherwise it's a bit tricky to figure out what's going on.

Reply:
Hopefully the confusing tests are well described now.